incubator/hnc/internal/validators/hncconfig.go

```diff
@@ -158,13 +159,17 @@ func (c *HNCConfig) checkConflictsForTree(gvk schema.GroupVersionKind, ao ancest
 	objs := ao.copy()
 	for _, o := range ns.GetSourceObjects(gvk) {
 		onm := o.GetName()
+		nsLabels := ns.GetLabels()
+		existingNs := objs[onm]
 		objs.add(onm, ns)
-		// If there are more than just the one (current) namespace we just added,
-		// there may be conflicts.
-		if objs.hasConflict(onm) {
-			// Currently the top conflicting ancestor would overwrite the object.
-			// TODO: check if this is a real conflict
-			conflicts = append(conflicts, fmt.Sprintf("  Object %q in namespace %q would overwrite the one in %q", onm, objs.top(onm), ns.Name()))
+		// Currently the top conflicting ancestor would overwrite the object.
+		// TODO: check if this is a real conflict
+		for _, nnm := range existingNs {
+			// check if the existing ns will propagate this object to the current ns
+			inst := c.Forest.Get(nnm).GetSourceObject(gvk, onm)
+			if ok, _ := selectors.ShouldPropagate(inst, nsLabels); ok {
+				conflicts = append(conflicts, fmt.Sprintf("  Object %q in namespace %q would overwrite the one in %q", onm, objs.top(onm), ns.Name()))
```

Comment:
I don't think `objs.top(onm)` is correct anymore?

Reply:
changed to `nnm`